src/main/java/org/assertj/core/api/AbstractClassLoaderAssert.java

```diff
@@ -0,0 +1,296 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2022 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.error.classloader.ShouldHaveParent.shouldHaveParent;
+import static org.assertj.core.error.classloader.ShouldHaveResource.shouldHaveResource;
+import static org.assertj.core.error.classloader.ShouldNotHaveParent.shouldNotHaveParent;
+import static org.assertj.core.error.classloader.ShouldNotHaveResource.shouldNotHaveResource;
+import static org.assertj.core.util.Strings.quote;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URL;
+import java.util.Enumeration;
+import org.assertj.core.internal.Failures;
+import org.assertj.core.internal.Objects;
+import org.assertj.core.util.CheckReturnValue;
+
+/**
+ * Base class for all implementations of assertions for {@link ClassLoader class loaders}.
+ *
+ * @param <SELF> the "self" type of this assertion class. Please read &quot;<a
+ *               href="http://bit.ly/1IZIRcY" target="_blank">Emulating 'self types' using Java
```

Comment:
Would it be better to put the full URL here?

Reply:
Yeah, I agree. This was reusing the documentation headers from existing code in this project, but I can update that. Good shout

Reply:
Updated